SchedulePlatform.Api/Controllers/AppointmentController.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using SchedulePlatform.Api.Models;
+using SchedulePlatform.Api.Mappings;
+using SchedulePlatform.Models.Entities;
+using SchedulePlatform.Service.Interfaces;
+
+namespace SchedulePlatform.Api.Controllers
+{
+	[Route("api/[controller]")]
+
+	[ApiController]
+
+	public class AppointmentController : ControllerBase
+	{
+		private readonly IAppointmentService _service;
+
+		public AppointmentController(IAppointmentService service)
+		{
+			_service = service;
+		}
+
+		[HttpGet]
+
+		public Appointment[] GetAppointments()
+		{
+			return _service.GetAll();
+		}
+
+		[HttpPost]
+
+		public Appointment AddAppointment(Appointment appointment)
+		{
+			return _service.Add(appointment);
+		}
+
+		[HttpGet("{id}")]
+
+		public IActionResult GetById(Guid id)
+		{
+			var appointmentResult = _service.GetById(id);
+
+			if (appointmentResult==null)
```

Comment:
I would say this kind of logic needs to stay in the service. And the service can throw itself an exception if the appointment is not found.